src/poc/miner_poc_statem.erl

```diff
@@ -252,20 +254,20 @@ receiving(info, {blockchain_event, {add_block, _Hash, _, _}}, #data{receiving_ti
         0 ->
             lager:warning("timing out, no receipts @ ~p", [_Hash]),
             %% we got nothing, no reason to submit
-            {next_state, requesting, save_data(Data#data{state=requesting})};
+            {next_state, requesting, save_data(maybe_init_addr_hash(Data#data{state=requesting}))};
         _ ->
             lager:warning("timing out, submitting receipts @ ~p", [_Hash]),
             try
                 Ref = submit_receipts(Data),
-                {next_state, waiting, save_data(Data#data{state=waiting, txn_ref=Ref})}
+                {next_state, waiting, save_data(maybe_init_addr_hash(Data#data{state=waiting, txn_ref=Ref}))}
             catch C:E:S ->
                     lager:warning("error ~p:~p submitting requests: ~p", [C, E, S]),
-                    {next_state, requesting, save_data(Data#data{state=requesting})}
+                    {next_state, requesting, save_data(maybe_init_addr_hash(Data#data{state=requesting}))}
             end
     end;
 receiving(info, {blockchain_event, {add_block, _Hash, _, _}}, #data{receiving_timeout=T}=Data) ->
     lager:info("got block ~p decreasing timeout", [_Hash]),
-    {keep_state, save_data(Data#data{receiving_timeout=T-1})};
+    {keep_state, save_data(maybe_init_addr_hash(Data#data{receiving_timeout=T-1}))};
```

Comment:
I feel like we should be able to do this just in load and init?

Reply:
No as the filter might need to be re-intited on a block height change.